src/NServiceBus.Core.Tests/Recoverability/RecoverabilityExecutorTests.cs

```diff
@@ -114,36 +86,32 @@ public void SetUp()
         }
 
         [Test]
-        public async Task When_delayed_retries_not_supported_but_policy_demands_it_should_move_to_error()
+        public void When_delayed_retries_not_supported_but_policy_demands_it_should_throw()
```

Comment:
I'm still on the fence when it comes to throwing. How about going with the safer approach which in my opinion is fallback to `MoveToError` and log warning?